feat(packages/angular): add new @lucide/angular package

karsa-mistmere opened the pull request:
## Description

Adds new standalone-first Angular package that better matches the ecosystem established by `lucide-react`:
- one standalone component per icon
- dynamic icon component with IconNode/name input that accepts:
  - icon components
  - icon nodes from the `lucide`/`@lucide/icons` packages
  - icon names for icons provided through `provideLucideIcons` via dependency injection
- default icon configuration provided through `provideLucideConfig` via dependency injection
- support for all configuration options established in earlier packages
- support for providing an accessible label via an `<svg:title>` element. Unless this accessible label is provided, the SVGs will automatically add `aria-hidden="true"`

## TODO
- [x] Add package documentation to site
- [x] Add more documentation to code
- [x] Explore the possibility of a migration schematic from `lucide-angular`

## TODO, but requires #3881
- [ ] Add `@lucide/angular/dynamic` entry point for dynamic icon imports using `@lucide/icons`
- [ ] Add `@lucide/angular/material` entry point for [registering icon data via `MatIconRegistry`](https://material.angular.dev/components/icon/overview#registering-icons)

## Before Submitting <!-- For every PR! -->
<!-- All of these requirements must be fulfilled. -->
- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

ericfennis reviewed (commented):
Nice work @karsa-mistmere, this looks very promising!
Some comments

karsa-mistmere commented:
> @karsa-mistmere Any updates on this? Can I help with something? I assume I need to fix the "next" tag deployment of `@lucide/icons` right?

Yeah, but first, I still want to add the aliases in `@lucide/icons`, as we discussed, I'm on it already!

karsa-mistmere commented:
@ericfennis: added some extensive rewrites:

- as per https://github.com/lucide-icons/lucide/pull/3897#discussion_r2640158469, I switched from separate node+name to self-describing `LucideIconData` objects similar to `@lucide/icons` – so no more toKebabCase hackery anywhere in the code apart from some legacy helpers, and as an added bonus there is now backwards compatible alias support for name based icon providers 🚀
- as per https://github.com/lucide-icons/lucide/pull/3897#discussion_r2840064045 I renamed `LucideIconComponentType` => `LucideIcon`, and `LucideIcon` => `LucideDynamicIcon`
- I added backwards compatible CSS class support as already implemented in `@lucide/icons`
- I switched to the `vector-effect: non-scaling-stroke` implementation from using computed stroke widths
- as requested, I added some unit tests on `LUCIDE_CONFIG` provider usage

ericfennis commented:
@karsa-mistmere Awesome work!
